README.md

```diff
@@ -2,51 +2,126 @@
+
+### General configuration
+
+* `core` — major testing dependency or method
+  Should have one of the following values:
+
+  * `tarantool` — test suite used for functional testing
+  * `app` — TAP-tests, another functional test suite
+  * `unittest` — unit testing test suite
+  * `luatest` — test suite using luatest library
+
+* `description` — test suite description
+* `script` (optional) — shebang file to start tarantool with.
+  Most often, the value is a file in the same directory as `suite.ini`, like `box.lua` or `master.lua`.
+  
+* `config` — name of a test configuration file, for example, `engine.cfg`.
+  TODO: what's a test configuration file?
+
+* `lua_libs` — paths to Lua files that should be copied to the directory,
+  where a server is started. For example:
+
+  ```ini
+  lua_libs = lua/require_mod.lua lua/serializer_test.lua lua/process_timeout.lua
+  ```
+
+### Skipping tests
+
+A number of fields are used to disable (skip) certain tests:
+
+* `disabled` — list of tests that should be skipped
+* `release_disabled` — list of tests that should be skipped when Tarantool is
+    built with `Release`
+* `valgrind_disabled` — list of tests that should be skipped when Valgrind is enabled
+* `long_run` — mark tests as long, such tests will run only with `--long` option.
+
+  ```ini
+  long_run = t1.test.lua t2.test.lua
+  ```
+
+### Other parameters
+
+
+* `show_reproduce_content` (optional, `True` by default) — TODO: what's this?
+* `is_parallel` (optional, `False` by default) — whether the tests in the suite can run in parallel
+
+* `use_unix_sockets` (optional, `False` by default) — use hard-coded UNIX sockets
+* `use_unix_sockets_iproto` (optional, `False` by default) — use hard-coded UNIX sockets for IProto.
+
+### Fragile tests
+
+Tests which fail sometimes due to external reasons can be marked as fragile (flaky).
+Test-run will retry each test if it fails.
+It's recommended to provide a list of related
+[tarantool/tarantool](https://github.com/tarantool/tarantool) issues
+next to each fragile test.
+
+```ini
+fragile = {
+    "retries": 10,
+    "tests": {
+        "tarantoolctl.test.lua": {
+            "issues": [ "gh-5059", "gh-5346" ],
+        },
+        {...},
+        {...},
+    }
+}
+```
+
+## Steps of a test run
+
+Each test consists of files `*.test(.lua|.sql|.py)?`, `*.result`,
+and an optional skip condition file `*.skipcond`.
+
+In the beginning of each test run, the `.skipcond` file is executed as a Python script.
+In the local Python environment there's a `self` object, which is an instance of the [`Test` class](./lib/test.py).
+To skip the test under some condition, set `self.skip = 1`.
+
+Next, the test in `.test(.lua|.py)?` is executed. 
+If this is a first test run and there's no `.result` file yet, it is generated from the output.
+If there is a `.result` file, output is written to a `.reject` file,
+which is then compared to `.result`.
```

Comment:
1. `.test.sql` as well.
2. And `_test.lua` in luatest test suites.
3. The autowritting `.result` file was removed together with introducing the `--update-result` option.

Reply:
4. There is no a word about TAP13 compatible output (it does not need a result file).

Reply:
I've removed enumeration of test file extensions from this line, it resolves (1) and (2).  Also, I won't mention writing `.result` files at all. Moved it to a separate issue: https://github.com/tarantool/test-run/issues/327

Added a few words about TAP13